### ISSUE TITLE
`--no-incremental` still writes `.mypy_cache`

### ISSUE THREAD

**Issue by asottile-sentry** [python/mypy#19489]
**Bug Report**

the help seems to indicate a cache will not be produced with `--no-incremental`:

```console
$ mypy --help | grep -A2 no-inc
  --no-incremental          Disable module cache (inverse: --incremental)
  --cache-dir DIR           Store module cache info in the given folder in
                            incremental mode (defaults to '.mypy_cache')
```

a workaround is `--cache-dir /dev/null` though this isn't super portable

**To Reproduce**

```python
print(1)
```

```
rm -rf .mypy_cache
mypy t.py --no-incremental
ls -ald .mypy_cache
```

**Expected Behavior**

```console
$ rm -rf .mypy_cache
$ mypy t.py --no-incremental
Success: no issues found in 1 source file
$ ls -ald .mypy_cache
ls: cannot access '.mypy_cache': No such file or directory
```

**Actual Behavior**

```console
$ rm -rf .mypy_cache
$ mypy t.py --no-incremental
Success: no issues found in 1 source file
$ ls -ald .mypy_cache
drwxr-xr-x 5 asottile wheel 160 Jul 22 07:36 .mypy_cache
```

**Your Environment**

- Mypy version used: 1.17.0
- Mypy command-line flags: `--no-incremental`
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used:

**Comment by emmatyping** [comment]
This is actually documented as the current behavior: https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-cache-dir

However I do agree having a portable flag to disable writing the cache is probably a good idea.

**Comment by asottile-sentry** [comment]
ah neat -- seems an opportunity to improve `--help` as well to match the docs (perhaps `s/in/for/` from the excerpt above would have been enough for me to dig further) and maybe "Disable reading module cache" for `--no-incremental`

**Comment by wyattscarpenter** [comment]
It's not relevant to your main request, but the documentation issue is yet another problem #12347 would fix. (But I wouldn't hold my breath for that)
